src/ImageSharp/Formats/Pbm/PbmConstants.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System.Collections.Generic;
+
+namespace SixLabors.ImageSharp.Formats.Pbm
+{
+    /// <summary>
+    /// Contains PBM constant values defined in the specification.
+    /// </summary>
+    internal static class PbmConstants
+    {
+        /// <summary>
+        /// The maximum allowable pixel value of a ppm image.
+        /// </summary>
+        public const ushort MaxLength = 65535;
```

Comment:
This is an unused. Is it needed?

Reply:
It should have been used to cap the MaxPixelValue, good catch ! Used it now.

Reply:
I'm curious about this constant. It appears to me that it is used for clamping during encoding but why not make `MaxPixelValue` in the metadata a `ushort`?

We're also automatically the value to 255 in the `PbmMetadata` constructor. I'm assuming that is to set the default to the non-wide color type.

Looking at how  `MaxPixelValue` is used it appears that any value over 255 indicates to use wider bit depth and we simply read/write the value in the header. I would suggest that instead of using this property in the metadata we should expand the `PbmColorType` type to add two new entries. `GrayscaleWide` and `RgbWide`.

Reply:
MaxPixelValue can be any value. For example, in GrayscalePlain it is set to 15.
If the pixels are left untouched, as I do in my implementation, the image would be rather dark. I deliberately left the upscaling of the pixel values up to the user of ImageSharp. Most application though upscale the pixel values to the full range of the data structure, when reading the image.

Because I chose to leave the pixel values untouched, I need to indicate to the user what scaling factor they need to apply, if they wish to do so. Hence the addition of MaxPixelValue to the Metadata.

Let me know what behavior you guys prefer.

Reply:
> Most application though upscale the pixel values to the full range of the data structure, when reading the image.

Wouldn't not doing this then mean that any application of a change to the image would render a different output to expected?

E.G

- Load Image
- Set [0,0] to color
- Save Image
- Load Image
- Read [0, 0]
- Compare to expected = Fail.

Would that be the correct summation? If so, we want to avoid people having to do any additional calculations when changing pixel values.

Reply:
As long as we do remain _consistently_ on the upscaling in both Encoding and Decoding, your example will work as expected.

As long as you stay within the range of [0..MaxPixelValue] the value will round trip, even in other applications (that do upscaling). 
If you go above the MaxPixelValue however, some other applications will refuse to load the image.

From this library I do not know what the intent of the user is. Some would want to retain the original pixel values. Some however, want the upscaling behavior they see in other applications.
By not upscaling automatically, I'm offering both alternatives (be it with an extra Mutate call for the second case).

Would you like to make this upscaling choice a property of IPbmEncoderOptions ?
And if so, what would be the default?

Reply:
I've implemented pixel value scaling. Didn't made it configurable, which aligns with other Formats. I users request configurability in the future, we can easily add it later.

Reply:
I'm still confused by this scaling stuff. 

We're reading the `MaxPixelValue` and then adjusting the brightness of the input pixels by that factor (_We're introducing parallelism and additional `TPixel/Vector4` conversion here which we've avoided elsewhere. It would be much better to operate on the byte row directly_).

We can now operate on the decoded pixels without additional per operation calculations - great!

However, when we encode, we pass the original `MaxPixelValue` back down to the encoder while leaving the original pixel values themselves unchanged. This means that subsequent decodes using our decoder would continue to run the upscaling each time which would be destructive. 

Please correct me if I'm wrong but as far as I can see `MaxPixelValue` has two functions. 
- To determine the bit depth of the encoded image
- To set the brightness of the image.

I still think the additional enum values are the best way to go with the correct scaling applied for decode/encode for each color/encoding type combination.

Reply:
My understanding of MaxPixelValue is the same as yours. I'll add some extra checks in the PbmRoundTripTest to make sure we're not messing with the MaxPixelValue.

The behavior I see in XnConvert and Irfanview, is what I would like to copy here. Decoded the pixels are upscaled to capacity for the pixel type. En during Encoding the scaled pixel values are written, together with a modified MaxPixelValue.

Now for the enums, I agree that enums are better then the raw MaxPixelValue form the image. Here, let's also look at [PAM](http://netpbm.sourceforge.net/doc/pam.html) images, as these are a likely addition lateron. PAM adds several pixel types, making a single enum a bit big. I therefore propose 2 enums:

1. ColorType, having the following values: BlackAndWhite, GrayScale, Rgb, Grayscale_Alpha, Rgb_Alpha.
2. PixelDataType (PAM calls these TypleTypes), having values: Bit, Byte, Short, Float and Double

Combining it into a single enum will give you 17 entries (B&W-Bit + 4 times 4 for the other combinations).

@JimBobSquarePants: Let me know your enum preference.

Reply:
17... Yeah best splitting them up. That's what we do with png.  We use [utilities](https://github.com/SixLabors/ImageSharp/blob/4ee7830105afa04629a1d9261ae26786a08584ef/src/ImageSharp/Formats/Png/PngEncoderOptionsHelpers.cs) to ensure valid combinations of the two enums are in use.

Reply:
@JimBobSquarePants  I have pushed the Enum changes, let me know your comments.